core-blocks/playlist/edit.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	InspectorControls,
+	MediaUpload,
+	BlockControls,
+	editorMediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.addFiles = this.addFiles.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			editing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	uploadFromFiles( event ) {
+		this.addFiles( event.target.files );
+	}
+
+	addFiles( files ) {
+		const { setAttributes } = this.props;
+		editorMediaUpload( {
+			allowedType: [ 'audio', 'video' ],
+			filesList: files,
+			onFileChange: ( media ) => {
+				if ( media && media[ 0 ].mime_type ) {
+					const type = media[ 0 ].mime_type.split( '/' )[ 0 ];
+					const ids = JSON.stringify( media.map( ( item ) => item.id ) );
+					setAttributes( { ids, type } );
+					this.setState( { editing: false } );
+				}
+			},
+		} );
+	}
+
+	toggleAttribute( attribute ) {
+		return ( newValue ) => {
+			this.props.setAttributes( { [ attribute ]: newValue } );
+		};
+	}
+
+	render() {
+		const { attributes, setAttributes, className } = this.props;
+		const { editing } = this.state;
+		const { tracklist, artists, images, style, type } = attributes;
+		const styleOptions = [
+			{ value: 'light', label: __( 'Light' ) },
+			{ value: 'dark', label: __( 'Dark' ) },
+		];
+
+		const onSelectMedia = ( media ) => {
+			//check if there are returned media items and set attributes when there are
+			if ( media && media[ 0 ].url ) {
+				const ids = JSON.stringify( media.map( ( item ) => item.id ) );
+				setAttributes( { tracklist, artists, images, ids, type: media[ 0 ].type } );
```

Comment:
I think `onSelectMedia` is a pretty clear name of the function. We could just remove the current one and rename `addFiles` to `onSelectMedia`

Reply:
We're admittedly very inconsistent with this, but I believe the current consensus within Gutenberg is to name functions (including `onEvent` callbacks) so that they

a) Begin with a verb
b) Describe _what_ the callback does, rather than what event it handles

Reply:
I went ahead and renamed `addFiles` to `onUploadFiles` to fit the naming convention. After comparing to `onSelectMedia` I remembered why they were separate. One of these handles the objects returned from the browser upload and the other handles the objects from selecting items from the Media modal. There might be a better way to do this but for the time being I'll at least get this named proper. Feel free to change this if you'd like. :) 
Also removed `tracklist`, `artists`, and `images` from `setAttributes` Those were there in anticipation of figuring out how to pass the settings back from the media modal. Now we hide the settings, so no need. Fixed and moving to other changes.